external/fv3fit/fv3fit/keras/adapters.py

```diff
@@ -108,3 +108,59 @@ def _rename_graph_outputs_to_match_output_keys(model: tf.keras.Model) -> tf.kera
     model_dict_out = tf.keras.Model(inputs=inputs, outputs=renamed)
     model_dict_out(inputs)
     return model_dict_out
+
+
+def _ensure_list_input(model: tf.keras.Model) -> tf.keras.Model:
+    """
+    Takes dict input + output model and converts it to take in
+    list inputs but keep original dict outputs. Useful for connecting
+    input layers with names different from the original model.
+    
+    Should NOT be used outside of helping rename_dict_input
+    in this module, as there is no way of specifying list order !!!
+    """
+    if isinstance(model.inputs, Mapping):
+        inputs = [input for input in model.inputs]
+    elif model.inputs is None:
+        raise ValueError(
+            f"Cannot detect inputs of model {model}. " "Custom models may not work."
+        )
+    else:
+        inputs = model.inputs
+    list_input_model = tf.keras.Model(inputs=inputs, outputs=model.outputs)
+    list_input_model(inputs)
+    # need to keep outputs dict- without calling this the model will revert
+    # to list outputs
+    return ensure_dict_output(list_input_model)
+
+
+def rename_dict_input(
+    model: tf.keras.Model, translation: Mapping[str, str]
+) -> tf.keras.Model:
+    """Rename the inputs of a dict-output model
+
+    Args:
+        model: a tensorflow model. must return dicts
+        transation: a mapping from old names (the keys returned by ``model``) to
+            the new output names.
+
+    """
+
+    # need to have list format inputs to connect renamed layers to rest of the model
+    list_input_model = _ensure_list_input(model)
```

Comment:
This is kind of wonky but was the only way I could figure out to connect the renamed input layers to the rest of the model if it had dict inputs and expected the old keys as input layer names.